service/frontend/workflowHandler.go

```diff
@@ -2176,27 +1827,47 @@ func (wh *WorkflowHandler) GetWorkflowExecutionHistory(
 		token.IsWorkflowRunning = isWorkflowRunning
 		token.PersistenceToken = nil
 	}
+	rawHistoryQueryEnabled := wh.config.EnableRawHistoryQuery() && wh.config.EnableRawHistoryQueryByDomain(getRequest.GetDomain())
 
 	history := &gen.History{}
+	historyBlob := []*gen.DataBlob{}
```

Comment:
if both are non-nil, how does the client know which history is valid ?

Reply:
> if both are non-nil, how does the client know which history is valid ?

fixed